DQM/src/DQM/SampleValidation.cxx

```diff
@@ -0,0 +1,130 @@
+      std::vector<int> parents_track_ids = p.getParents();
+      std::vector<int> daughters = p.getDaughters();
+
+      for (auto const& parent_track_id: parents_track_ids) {
+	if (parent_track_id == 0) {
+	  histograms_.fill("pdgid_primaries", pdgid_label(pdgid));
+	  histograms_.fill("energy_primaries", energy);
+	  hard_thresh = (2500/4000)*energy;
+	  primary_daughters = daughters;
+	  for (const ldmx::SimTrackerHit &sphit : targetSPHits) {
+	    if (sphit.getTrackID() == it.first && sphit.getPosition()[2] < 0) {
+	      histograms_.fill("beam_smear", vertex[0], vertex[1]);
+	    }
+	  }
+	}
+      }
+    }
+
+    std::vector<std::vector<int>> hardbrem_daughters;
+
+    for (auto const& it : particle_map) {
+      int trackid = it.first;
+      ldmx::SimParticle p = it.second;
+      for (auto const& primary_daughter : primary_daughters) {
+	if (trackid == primary_daughter) {
+	  histograms_.fill("pdgid_primarydaughters", pdgid_label(p.getPdgID()));
+	  if (p.getPdgID() == 22) {
+	    histograms_.fill("energy_daughterphoton", p.getEnergy());
+	  }
+	  if (p.getEnergy() >= hard_thresh) {
+	    histograms_.fill("pdgid_harddaughters", pdgid_label(p.getPdgID()));
+	    histograms_.fill("startZ_hardbrem", p.getVertex()[2]);
+	    histograms_.fill("endZ_hardbrem", p.getEndPoint()[2]);
+	    histograms_.fill("energy_hardbrem", p.getEnergy());
+	    hardbrem_daughters.push_back(p.getDaughters());
+	  }
+	}
+      }
+    }
+
+    for (auto const& it : particle_map) {
+      int trackid = it.first;
+      ldmx::SimParticle p = it.second;
+      for (const std::vector<int> &daughter_track_id : hardbrem_daughters){
+	for (const int &daughter_id : daughter_track_id) {
+	  if (trackid == daughter_id) {
+	    histograms_.fill("pdgid_hardbremdaughters", pdgid_label(p.getPdgID()));
+	    histograms_.fill("startZ_hardbremdaughters", p.getVertex()[2]);
+	  }
+	}
+      }
+    }
+
+
+    return;
+  }
+
+  int SampleValidation::pdgid_label(const int pdgid) {
+    int label = 18; // initially assign label as "anything else"/overflow value, only change if the pdg id is something of interest
+    if (pdgid == -11) label = 1; // e+
+    
+    if (pdgid == 11) label = 2; // e-
+    
+    if (pdgid == -13) label = 3; // μ+
+    
+    if (pdgid == 13) label = 4; // μ-
+    
+    if (pdgid == 22) label = 5; // γ
+    
+    if (pdgid == 2212) label = 6; // proton
+    
+    if (pdgid == 2112) label = 7; // neutron
+    
+    if (pdgid == 211) label = 8; //π+
+    
+    if (pdgid == -211) label = 9; //π-
+    
+    if (pdgid == 111) label = 10; //π0
+    
+    if (pdgid == 321) label = 11; // K+
+    
+    if (pdgid == -321) label = 12; // K-
+    
+    if (pdgid == 130) label = 13; // K-Long
+    
+    if (pdgid == 310) label = 14; // K-Short
+    
+    if (pdgid == 3122 || pdgid == 3222 || pdgid == 3212 || pdgid == 3112 || pdgid == 3322 || pdgid == 3312) label = 16; // strange baryon
+    
+    if (pdgid > 10000) label = 15; //nuclei
```

Comment:
Is it worth differentiating between general nuclei and light ions here? The production of alpha, deuterium, and tritium from PN interactions is non-trivial.

Reply:
I can include this distinction since I meant for the label to mainly catch remnants of a tungsten explosion. Do you think "light"/"heavy" nuclei is granular enough, with "light" being only hydrogen or helium nuclides?

Reply:
The bertini cascade will attempt to take nucleons being ejected from the cascade that are "similar" and bunch them together into deuterium, tritium, alpha (called "Coalescence") so you can get these even for quite ordinary interactions. 

There is a version of a test for what counts as a light ion here (that probably should live somewhere else...)
https://github.com/LDMX-Software/ldmx-sw/blob/8e6e9e195eb71ad78f7aa94648a179342a12e1f6/Biasing/include/Biasing/PhotoNuclearTopologyFilters.h#L62C1-L78